ksp_plugin/planetarium.cpp

```diff
@@ -268,6 +272,101 @@ RP2Lines<Length, Camera> Planetarium::PlotMethod2(
   return lines;
 }
 
+void Planetarium::PlotMethod3(
+    Trajectory<Barycentric> const& trajectory,
+    Instant const& first_time,
+    Instant const& last_time,
+    Instant const& now,
+    bool reverse,
+    std::function<void(ScaledSpacePoint const&)> add_point,
+    int const max_points,
+    Length* minimal_distance) const {
+  double const tan²_angular_resolution =
+      Pow<2>(parameters_.tan_angular_resolution_);
+  auto const final_time = reverse ? first_time : last_time;
+  auto previous_time = reverse ? last_time : first_time;
+
+  if (minimal_distance != nullptr) {
+    *minimal_distance = Infinity<Length>;
+  }
+
+  Sign const direction = reverse ? Sign::Negative() : Sign::Positive();
+  if (direction * (final_time - previous_time) <= Time{}) {
+    return;
+  }
+  RigidMotion<Barycentric, Navigation> to_plotting_frame_at_t =
+      plotting_frame_->ToThisFrameAtTime(previous_time);
+  DegreesOfFreedom<Navigation> const initial_degrees_of_freedom =
+      to_plotting_frame_at_t(
+          trajectory.EvaluateDegreesOfFreedom(previous_time));
+  Position<Navigation> previous_position =
+      initial_degrees_of_freedom.position();
+  Velocity<Navigation> previous_velocity =
+      initial_degrees_of_freedom.velocity();
+  Time Δt = final_time - previous_time;
+
+  add_point(plotting_to_scaled_space_(previous_position));
```

Comment:
In the previous code we would sometimes return 0 segments.  Here we always return at least 1 point.  Do we care?

Reply:
No; the handling of segments is done by the GPU which gets the vertices and indices.